django/http/request.py

```diff
@@ -432,8 +438,16 @@ def __init__(self, query_string=None, mutable=False, encoding=None):
             except UnicodeDecodeError:
                 # ... but some user agents are misbehaving :-(
                 query_string = query_string.decode('iso-8859-1')
-        for key, value in limited_parse_qsl(query_string, **parse_qsl_kwargs):
-            self.appendlist(key, value)
+        try:
+            for key, value in parse_qsl(query_string, **parse_qsl_kwargs):
+                self.appendlist(key, value)
+        except ValueError as e:
+            if str(e) != 'Max number of fields exceeded':
```

Comment:
This feels a bit iffy to me and I would not consider the message a stable API. What other `ValueError` could be thrown here that makes this necessary?

Reply:
When updating the backported function, there was new code added that includes an additional `ValueError`. We are not using the new `strict_parsing` flag, however, so we could just assume the that `ValueError` raised is not from the branch guarded by that boolean flag?